src/libuvc/libuvc.cpp

```diff
@@ -402,42 +407,173 @@ namespace librealsense
             }
             power_state get_power_state() const override { return _state; }
 
-            void init_xu(const extension_unit& xu) override {
-              // TODO : implement.
-            }
+            void init_xu(const extension_unit& xu) override {};
             bool set_xu(const extension_unit& xu, uint8_t control, const uint8_t* data, int size) override
             {
-              // TODO : implement.
+                uvc_set_ctrl(_device_handle, xu.unit, control, (void *)data, size);
                 return true;
             }
             bool get_xu(const extension_unit& xu, uint8_t control, uint8_t* data, int size) const override
             {
-              // TODO : implement.
-                return true;
+                int status =
+                        uvc_get_ctrl(_device_handle, xu.unit, control, data, size, UVC_GET_CUR);
+
+                if ( status >= 0) {
+                    return true;
+                }
+                return false;
             }
             control_range get_xu_range(const extension_unit& xu, uint8_t control, int len) const override
             {
-              // TODO : implement.
-                control_range result{};
+                int status;
+                int32_t value;
+                int max;
+                int min;
+                int step;
+                int def;
+                status = uvc_get_ctrl(_device_handle, xu.unit, control, &max, sizeof(int32_t), UVC_GET_MAX);
```

Comment:
In case of an error thrown the message should include the attribute for which the query failed.
It also seem to be a case for refactoring to bind all four operation into a for(...) loop iterating through the properties. For consideration

Reply:
Done